pkg/scorecard.go

```diff
@@ -115,11 +139,35 @@ func RunScorecard(ctx context.Context,
 		Date: time.Now(),
 	}
 	resultsCh := make(chan checker.CheckResult)
-	go runEnabledChecks(ctx, repo, &ret.RawResults, checksToRun, repoClient, ossFuzzRepoClient,
+
+	// Set metadata for all checks to use. This is necessary
+	// to create remediations from the probe yaml files.
+	ret.RawResults.Metadata = map[string]string{
+		"repository.host":          repo.Host(),
+		"repository.name":          strings.TrimPrefix(repo.URI(), repo.Host()+"/"),
+		"repository.uri":           repo.URI(),
+		"repository.sha1":          commitSHA,
+		"repository.defaultBranch": defaultBranch,
+	}
+
+	go runEnabledChecks(ctx, repo, &ret.RawResults, checksToRun,
+		repoClient, ossFuzzRepoClient,
 		ciiClient, vulnsClient, resultsCh)
 
 	for result := range resultsCh {
 		ret.Checks = append(ret.Checks, result)
 	}
+
+	// Run the probes.
+	var findings []finding.Finding
+	// TODO(#3049): only run the probes for checks.
+	// NOTE: We will need separate functions to support:
+	// - `--probes X,Y`
+	// - `--check-definitions-file path/to/config.yml`
+	findings, err = zrunner.Run(&ret.RawResults, probes.All)
+	if err != nil {
+		return ScorecardResult{}, err
+	}
```

Comment:
If the probes have an error this will also affect non-probe users.

Reply:
Good point. I think we can discard the error. The errors will all be part of the findings so we don't need the returned err, see https://github.com/ossf/scorecard/blob/main/probes/zrunner/runner.go#L34-L45

Ill add a comment to explain why we discard the err.

Let me know if this is enough.